arch/x86/core/early_serial.c

```diff
@@ -11,10 +11,11 @@
 #include <soc.h>
 
 
-#define UART_IS_IOPORT_ACCESS \
-	DT_NODE_HAS_PROP(DT_CHOSEN(zephyr_console), io_mapped)
+#if DT_PROP_OR(DT_CHOSEN(zephyr_console), io_mapped, 0) != 0
+#define UART_IS_IOPORT_ACCESS 1
```

Comment:
Do we need to define it to `1` ?

Reply:
Can be also done with: 
define UART_IS_IOPORT_ACCESS (DT_PROP_OR(DT_CHOSEN(zephyr_console), io_mapped, 0) != 0) ? 1 : 0

Reply:
seriously it's fine as it is. Let's move on